streaming/receiver.c

```diff
@@ -667,8 +701,9 @@ static int rrdpush_receive(struct receiver_state *rpt)
         error("STREAM %s [receive from [%s]:%s]: cannot set timeout for socket %d", rrdhost_hostname(rpt->host), rpt->client_ip, rpt->client_port, rpt->fd);
 
     // convert the socket to a FILE *
-    FILE *fp = fdopen(rpt->fd, "r");
-    if(!fp) {
+    FILE *fp_out = fdopen(rpt->fd, "w");
+    FILE *fp_in = fdopen(rpt->fd, "r");
+    if(!fp_in) {
         log_stream_connection(rpt->client_ip, rpt->client_port, rpt->key, rpt->host->machine_guid, rrdhost_hostname(rpt->host), "FAILED - SOCKET ERROR");
         error("STREAM %s [receive from [%s]:%s]: failed to get a FILE for FD %d.", rrdhost_hostname(rpt->host), rpt->client_ip, rpt->client_port, rpt->fd);
```

Comment:
```sh
*** CID 357338:  Resource leaks  (RESOURCE_LEAK)
/streaming/receiver.c: 710 in rrdpush_receive()
704         FILE *fp_out = fdopen(rpt->fd, "w");
705         FILE *fp_in = fdopen(rpt->fd, "r");
706         if(!fp_in) {
707             log_stream_connection(rpt->client_ip, rpt->client_port, rpt->key, rpt->host->machine_guid, rrdhost_hostname(rpt->host), "FAILED - SOCKET ERROR");
708             error("STREAM %s [receive from [%s]:%s]: failed to get a FILE for FD %d.", rrdhost_hostname(rpt->host), rpt->client_ip, rpt->client_port, rpt->fd);
709             close(rpt->fd);
>>>     CID 357338:  Resource leaks  (RESOURCE_LEAK)
>>>     Variable "fp_out" going out of scope leaks the storage it points to.
710             return 0;
711         }
712     
713         rrdhost_wrlock(rpt->host);
714     /* if(rpt->host->connected_senders > 0) {
715             rrdhost_unlock(rpt->host);

** CID 357337:  Resource leaks  (RESOURCE_LEAK)
/libnetdata/popen/popen.c: 281 in popene_internal()
```

Reply:
I think this is also by design. We need it to go out of the function.

Reply:
I ran coverity again, and it did not find issues. I am going to write a report with tests I ran and approve the PR.